transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1118,6 +1142,20 @@ private void callHandlerCallbackLater(AbstractChannelHandlerContext ctx, boolean
         }
     }
 
+    private void wakeUpForAutoFlush() {
+        if (autoFlush && ENQUEUE_WAKEUP_TASK_UPDATER.compareAndSet(this, 0, 1)) {
+            if (null == wakeupTask) {
+                wakeupTask = new Runnable() {
+                    @Override
+                    public void run() {
+                        ENQUEUE_WAKEUP_TASK_UPDATER.set(DefaultChannelPipeline.this, 0);
+                    }
+                };
+            }
+            channel().eventLoop().execute(wakeupTask);
+        }
+    }
+
```

Comment:
…and here's a much more helpful stack trace:

```
Jun 20, 2016 10:12:57 PM io.netty.channel.ChannelOutboundBuffer safeSuccess
WARNING: Failed to mark a promise as success because it has failed already: DefaultChannelPromise@77e6c94f(failure: java.util.concurrent.RejectedExecutionException: event executor terminated), unnotified cause java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:799)
	at io.netty.channel.SingleThreadEventLoop.onEventLoopIteration(SingleThreadEventLoop.java:95)
	at io.netty.channel.AbstractChannel$AutoFlushTask.onWrite(AbstractChannel.java:1102)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.write(AbstractChannel.java:810)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.write(DefaultChannelPipeline.java:1309)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:722)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:714)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:800)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:707)
	at io.netty.handler.ssl.SslHandler.finishWrap(SslHandler.java:580)
	at io.netty.handler.ssl.SslHandler.wrap(SslHandler.java:566)
	at io.netty.handler.ssl.SslHandler.flush(SslHandler.java:500)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:760)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:752)
	at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:733)
	at io.netty.channel.DefaultChannelPipeline.flush(DefaultChannelPipeline.java:973)
	at io.netty.channel.AbstractChannel.flush(AbstractChannel.java:237)
	at io.netty.channel.AbstractChannel$AutoFlushTask.run(AbstractChannel.java:1091)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasksFrom(SingleThreadEventExecutor.java:356)
	at io.netty.channel.SingleThreadEventLoop.afterRunningAllTasks(SingleThreadEventLoop.java:140)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:374)
	at io.netty.channel.SingleThreadEventLoop.runAllTasks(SingleThreadEventLoop.java:130)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:393)
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:827)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:145)
	at java.lang.Thread.run(Thread.java:745)

Jun 20, 2016 10:12:57 PM io.netty.util.concurrent.SingleThreadEventExecutor runAllTasksFrom
WARNING: A task raised an exception.
java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:799)
	at io.netty.channel.SingleThreadEventLoop.onEventLoopIteration(SingleThreadEventLoop.java:95)
	at io.netty.channel.AbstractChannel$AutoFlushTask.run(AbstractChannel.java:1094)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasksFrom(SingleThreadEventExecutor.java:356)
	at io.netty.channel.SingleThreadEventLoop.afterRunningAllTasks(SingleThreadEventLoop.java:140)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:374)
	at io.netty.channel.SingleThreadEventLoop.runAllTasks(SingleThreadEventLoop.java:130)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:393)
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:827)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:145)
	at java.lang.Thread.run(Thread.java:745)
```